meerk40t/core/elements.py

```diff
@@ -3511,13 +4078,24 @@ def element_polygon(args=tuple(), **kwargs):
         )
         def element_polyline(command, channel, _, args=tuple(), **kwargs):
```

Comment:
The help for polyline still reads `(float, float)*` and likely is (x y)* for the more correct value there.

Reply:
Have changed that - on another note:
how does command trailing work, I though one command would look at all the parameters it would require and the remainder of the line would then take the result of that command use its as the base for further commands.
If you do have a command like polygon or polyline with an arbitrary amount of parameters this approach fails.
Any idea how to overcome that? Is there a pipe / end of command parameter in place that forces an end of parameters for a command?
(And still the way of generating the polyline / polygon elements needs to be amended  - will do this in this PR)

Reply:
The `@console_command` will uses a dictionary for everything that wasn't matched and one of the parameters it gives you is `args` this is all the arguments for the command without parsing. So if a command is not intended to be chained like `polygon` and `polyline` they can accept `args` which is just the tuple of split values in the `remainder` without context. The `remainder` is somewhat similar in that it's the string of `remaining` unparsed values after or within the current command.

The chaining of the commands is predicated on consuming all the commands successfully. So if we have a command that has 4 arguments then we know that `command a0 a1 a2 a3 <new_command>` is valid we could do chain our `shape` command into a `grid` command or add a `fill` or `stroke` as part of the current command. `circle 3in 3in 1in fill red` will chain the circle command which returns an `elements` out_type with `fill` which accepts an `elements` in_type. So we make a circle (adding it to the scene) and then give that circle a `red` fill value.

For some commands we don't know how many arguments we will have and in that case, chaining like this becomes impossible. If we use `polyline` we can't set a fill value at the end.

Reply:
It might be possible to run some validation on arguments and if the casting of the value causes a ValueError stop and call that arguments and could be added in as nargs="*" or so, assuming the type isn't `str` or something where all the values would be valid there.